api/v1alpha1/instancegroup_types.go

```diff
@@ -392,6 +395,10 @@ func (ig *InstanceGroup) GetUpgradeStrategy() *AwsUpgradeStrategy {
 func (ig *InstanceGroup) SetUpgradeStrategy(strategy AwsUpgradeStrategy) {
 	ig.Spec.AwsUpgradeStrategy = strategy
 }
+func (ig *InstanceGroup) Locked() bool {
+	annotations := ig.GetAnnotations()
+	return annotations[UpgradeLockedAnnotationKey] == "true"
```

Comment:
You can probably use `HasAnnotation` from https://github.com/keikoproj/instance-manager/blob/c0eb25d65856ca1261f70465c56950f4bfc0cdc3/controllers/providers/kubernetes/utils.go#L119

It has a bit more safety for case, etc.

Reply:
Same as above, v1alpha is included by `kubernetes/utils.go` so creates a circular dependency.

Reply:
I've duplicated the functionality into the `Locked()` method.

Reply:
👍